src/app/utils/field-validation/validators/decimalValidator.ts

```diff
@@ -0,0 +1,63 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { flow } from 'fp-ts/lib/function'
+import isFloat from 'validator/lib/isFloat'
+import isInt from 'validator/lib/isInt'
+
+import { IDecimalField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+import { ISingleAnswerResponse } from 'src/types/response'
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type DecimalValidator = ResponseValidator<ISingleAnswerResponse>
+type DecimalValidatorConstructor = (
+  decimalField: IDecimalField,
+) => DecimalValidator
+
+const makeDecimalFloatRangeValidator: DecimalValidatorConstructor = (
+  decimalField,
+) => (response) => {
+  const { customMin, customMax } = decimalField.ValidationOptions || {}
```

Comment:
is the fallback to empty object truly necessary?

Reply:
shouldn't be since customMin and customMax both have defaults, but we should also double check (eg by deleting ValidationOptions on a decimal field and checking if mongoose puts it back automatically)

Reply:
removed the fallback

Reply:
@liangyuanruo @mantariksh actually can we keep the fallback since there is really no downside; although mongoose restores the default when I tested it manually, the failing tests make me nervous